query-projector/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#[macro_use]
+extern crate error_chain;
+extern crate rusqlite;
+
+extern crate mentat_core;
+extern crate mentat_db;                 // For value conversion.
+extern crate mentat_query;
+extern crate mentat_query_algebrizer;
+extern crate mentat_query_sql;
+extern crate mentat_sql;
+
+use std::iter;
+
+use rusqlite::{
+    Row,
+    Rows,
+};
+
+use mentat_core::{
+    TypedValue,
+};
+
+use mentat_db::{
+    SQLValueType,
+    TypedSQLValue,
+};
+
+use mentat_query::{
+    Element,
+    FindSpec,
+    PlainSymbol,
+    Variable,
+};
+
+use mentat_query_algebrizer::{
+    AlgebraicQuery,
+    DatomsColumn,
+    QualifiedAlias,
+    /*
+    ConjoiningClauses,
+    DatomsTable,
+    SourceAlias,
+    */
+};
+
+use mentat_query_sql::{
+    ColumnOrExpression,
+    /*
+    Constraint,
+    FromClause,
+    */
+    Name,
+    Projection,
+    ProjectedColumn,
+    /*
+    SelectQuery,
+    TableList,
+    */
+};
+
+error_chain! {
```

Comment:
Huh, interesting, instead of `errors.rs`.  Not sure if I prefer this inline our in its own file right now.  Roll on.